frontend/src/components/home/measurements/client/ClientsAndProjectsEvolution.tsx

```diff
@@ -0,0 +1,142 @@
+import { Row } from "react-bootstrap";
+import ClientsAndProjectsEvolutionChart from "./ClientsAndProjectsEvolutionChart";
+import ClientsAndProjectsEvolutionList from "./ClientsAndProjectsEvolutionList";
+import { Container } from "react-bootstrap";
+import { Col } from "react-bootstrap";
+import { useState } from "react";
+import moment, { Moment } from "moment";
+import { useSelector } from "react-redux";
+import { ConfacState } from "../../../../reducers/app-state";
+import { t } from "../../../utils";
+import { Link } from "react-router-dom";
+import PeriodPicker from "../PeriodPicker";
+
+export interface clientChartData {
+  year: number;
+  monthIndex: number;
+  month: string;
+  clients: number | null;
+  clientsWithProjects: number | null;
+}
+const ClientsAndProjectsEvolution = () => {
+  const [from, setFrom] = useState<moment.Moment>(moment());
+  const [to, setTo] = useState<moment.Moment>(from.clone().add(11, "month"));
+  const models = useSelector((state: ConfacState) => ({
+    clients: state.clients,
+    projects: state.projects,
+  }));
+
+  let interim = from.clone();
+  let timeValues: Moment[] = [];
+
+  while (interim.startOf("month").isSameOrBefore(to.endOf("month"))) {
+    timeValues.push(interim.clone());
+    interim.add(1, "month");
+  }
+
+  const dataSet: clientChartData[] = timeValues.map((date) => {
+    return {
+      year: date.year(),
+      monthIndex: date.month(),
+      month: date.format("MMM"),
+      clients: null,
+      clientsWithProjects: null,
+    };
+  });
+
+  const clientsWithProject = models.clients.map((client) => {
+    return {
+      detail: client,
+      projects: models.projects.filter(
+        (project) => project.client.clientId === client._id
+      ),
+    };
+  });
+
+  let count: number = 0;
+  dataSet.forEach((entry) => {
+    let amount: number = models.clients
+      .filter(
+        (client) => moment.utc(client.audit.createdOn).year() === entry.year
+      )
+      .filter(
+        (client) =>
+          moment.utc(client.audit.createdOn).month() === entry.monthIndex
+      ).length;
+    count += amount;
+    entry.clients = count;
+  });
+
+  /* Here I check if a client has a project with a start and enddate 
+  that is valid for the month and year pairs in the dataset and count them up for the year and month pairs. */
+
+  dataSet.forEach((entry) => {
+    let amount: number = clientsWithProject.filter(
+      (client) =>
+        client.projects.length > 0 &&
+        client.projects.some((project) =>
+          moment([entry.year, entry.monthIndex]).isBetween(
+            project?.startDate,
+            project?.endDate
+          )
+        )
+    ).length;
+    entry.clientsWithProjects = amount;
+  });
```

Comment:
`forEach`: probeer dit te vermijden!
--> Probeer meer functioneel/side-effect free te denken/programmeren

Let ook op, meerdere keren over collecties gaan loopen, dat kost tijd.
Zal tijdens onze meeting effe duiden:
--> Je hoeft daar niet teveel rekening met te houden, maar wel iets om in het achterhoofd te houden